utils/queryApiGateway.ts

```diff
@@ -176,6 +176,16 @@ export default async function queryApiGateway(req: IncomingMessage, uniqueNumber
   } catch (error) {
     const logger: Logger = Logger.getInstance()
     logger.log('error', error, 'API gateway error')
+    throw error
+  }
+
+  if (apiData?.uniqueNumber !== uniqueNumber) {
+    const mismatchError = new Error(
+      `Mismatched API response and Header unique number (${apiData.uniqueNumber || 'null'} and ${uniqueNumber})`,
+    )
+    const logger: Logger = Logger.getInstance()
```

Comment:
just curious, why do we declare new instances of Logger each time we call logger.log in this file / would it be bad to just declare it once in the outer scope?

Reply:
it handles that! Rocket implemented the logger as a singleton, so `getInstance` will get a new one only if it hasnt aready been invoked, otherwise it retrieves the existing one

https://github.com/cagov/ui-claim-tracker/blob/ed59fd1e2b33cafc41fb709143f457e39ac012fb/utils/logger.ts#L35-L43

Reply:
Oh cool, makes sense! Still curious if it would be bad (whether for readability or functionality) to just declare it once in the outer scope, maybe in a beforeAll or something. Maybe also a question for @rocketnova

Reply:
@kalvinwang  👀 What is a `beforeAll` in this context? 

There isn't really any harm in calling `const logger: Logger = Logger.getInstance()` at the top of `queryApiGateway()` but since we only need it in three edge cases, I figured it was minimal enough to only call as necessary.

Reply:
gotcha ok that makes sense, just checking! in jest, beforeAll is like beforeEach, but it's only run once instead of each time

Reply:
we're in a utility file though, not a test file